### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by xh3b4sd**
This PR sets up the repository and provides the base cluster object. The structure is basically copy/paste from https://github.com/giantswarm/kvmtpr. I only added the `yaml` tags to the struct properties.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package node
+
+type Node struct {
```
Comment (iaguis) [inline node/node.go:3]: What about `Hostname`?

File: docker/docker.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package docker
+
+type Docker struct {
+	Registry Registry `json:"registry" yaml:"registry"`
+	Daemon   Daemon   `json:"daemon" yaml:"daemon"`
```
Comment (iaguis) [inline docker/docker.go]: I know we should move away from this later, but I'm missing `ImageNamespace`

File: etcd/etcd.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package etcd
+
+type Etcd struct {
+	Domain string `json:"domain" yaml:"domain"`
+	IP     string `json:"ip" yaml:"ip"`
+	Port   string `json:"port" yaml:"port"`
```
Comment (iaguis) [inline etcd/etcd.go:6]: `Prefix`?

File: node/node.go (hunk 1 above)
Comment (xh3b4sd) [reply node/node.go:3]: Done.

File: docker/docker.go (hunk 2 above)
Comment (xh3b4sd) [reply docker/docker.go]: Done.

File: etcd/etcd.go (hunk 3 above)
Comment (xh3b4sd) [reply etcd/etcd.go:6]: Done.

**Comment by xh3b4sd** [comment]
@iaguis good catch. I totally missed the fields because they have been introduced in the `awstpr`. Really good to split that up so we will not have to deal with issues like this in the future.

**Review by iaguis — APPROVED** [review]
lgtm. I like this :)

**Pull request merged**